lib/V8/v8-globals.h

```diff
@@ -485,21 +498,18 @@ struct TRI_v8_global_t {
   /// @brief wrapper around a v8::Persistent to hold a shared_ptr and cleanup
   class SharedPtrPersistent {
    public:
-    SharedPtrPersistent(                    // constructor used ONLY by
-                                            // SharedPtrPersistent::emplace(...)
-        v8::Isolate& isolate,               // isolate
-        std::shared_ptr<void> const& value  // value
-    );
+    // emplace persistent shared pointer
+    static std::pair<SharedPtrPersistent&, bool> emplace(
+        v8::Isolate& isolate, std::shared_ptr<void> const& value);
+
+    // constructor used ONLY by SharedPtrPersistent::emplace(...)
```

Comment:
Shall we make the constructor private then?

Reply:
Unfortunately it doesn't compile otherwise :-(